### PR TITLE
Load thpres from restart file

### PR DESCRIPTION
**Pull request by joakim-hove**
See opm/opm-common#410

### CODE REVIEW

File: ebos/eclproblem.hh [hunk 1]
```diff
@@ -851,6 +849,10 @@ public:
         return thresholdPressures_;
     }
 
+    EclThresholdPressure<TypeTag>& thresholdPressure() {
```
Comment (andlaus) [inline ebos/eclproblem.hh]: nit: the eWoms convention for oneline methods is:
```c++
EclThresholdPressure<TypeTag>& thresholdPressure()
{ return thresholdPressures_; }
```

(this repeats below.)

File: ebos/eclwriter.hh [hunk 2]
```diff
@@ -260,6 +260,13 @@ public:
             unsigned globalIdx = collectToIORank_.localIdxToGlobalIdx(elemIdx);
             eclOutputModule_.setRestart(restart_values.solution, elemIdx, globalIdx);
         }
+        const auto& input_thpres = eclState().getSimulationConfig().getThresholdPressure();
```
Comment (andlaus) [inline ebos/eclwriter.hh]: camelCase please. (same for `restartValues`, `mutableSimulator` and `thpresValues`)

**Comment by joakim-hove** [comment]
OK - coding style updated.

**Comment by andlaus** [comment]
cool, thanks a lot. merge whenever convenient. BTW: is this jenkinsed as part of OPM/opm-simulators#1488?

**Comment by joakim-hove** [comment]
> is this jenkinsed as part of OPM/opm-simulators#1488?

Yes

**Comment by joakim-hove** [comment]
OK - this has been updated due to failures in the integration tests. Observe the member vaiable `restartApplied` in the `eclproblem.hh` implementation: My guess is that that is introduced specifically for `THPRES` - and should therefore not be required any longer; however the current implementation still needs the `restartApplied`flag to avoid a recursive death.

**Comment by totto82** [comment]
> My guess is that that is introduced specifically for THPRES - and should therefore not be required any longer; however the current implementation still needs the restartAppliedflag to avoid a recursive death.

The flag can be removed together with the recursive call. I can clean this up after this is merged if you don't want to do it yourself.

**Comment by joakim-hove** [comment]
> The flag can be removed together with the recursive call. I can clean this up after this is merged if you don't want to do it yourself.

Cool if you do that - my initial attempt just led to a recursive meltdown.

**Pull request merged**